ehsm_kms_service/secret_manager_apis.js

```diff
@@ -265,9 +360,709 @@ const createSecret = async (res, appid, payload, DB) => {
+                res.send(_result(400, 'forceDelete must be true or false'))
+                return
+            }
+        }
+        //Set default delete time
+        if (recoveryPeriod == '' || recoveryPeriod == undefined) {
+            recoveryPeriod = DEFAULT_DELETE_RECOVERY_DAYS
+        }
+
+        if (forceDelete == 'true') {
+            //Force delete secret metadata and secret version data by appId and secretName
+            if (await forceDeleteData(DB, appid, secretName)) {
+                res.send(_result(200, `The ${base64_decode(secretName)} delete success`))
+                return
+            } else {
+                res.send(_result(400, 'force delete failed'))
+                return
+            }
+        } else {
+            //change delete secret metadata and secret version data according to the scheduled deletion time
+            //Query and change the secret metadata
+            const query_matadata = {
+                selector: {
+                    appid,
+                    secretName
+                },
+                fields: [
+                    '_id',
+                    '_rev',
+                    'appid',
+                    'secretName',
+                    'encryptionKeyId',
+                    'description',
+                    'createTime',
+                    'deleteTime',
+                    'plannedDeleteTime',
+                    'rotationInterval',
+                    'lastRotationDate',
+                    'nextRotationDate'
+                ],
+                limit: 1
+            }
+            let secret_metadata_res = await DB.partitionedFind('secret_metadata', query_matadata)
+            if (secret_metadata_res.docs.length > 0) {
+                secret_metadata_res.docs[0].deleteTime = deleteTime
+                secret_metadata_res.docs[0].plannedDeleteTime = deleteTime + 24 * 60 * 60 * 1000 * recoveryPeriod
+                //Query and change the secret version data
+                const query_version_data = {
+                    selector: {
+                        appid,
+                        secretName,
+                        deletedFlag: false
+                    },
+                    fields: [
+                        '_id',
+                        '_rev',
+                        'appid',
+                        'secretName',
+                        'versionId',
+                        'deletedFlag',
+                        'secretData',
+                        'createTime',
+                        'versionStage',
+                    ]
+                }
+                let secret_version_data_res = await DB.partitionedFind('secret_version_data', query_version_data)
+                if (secret_version_data_res.docs.length > 0) {
+                    for (var i = 0; i < secret_version_data_res.docs.length; i++) {
+                        secret_version_data_res.docs[i].deletedFlag = true
```

Comment:
I reviewed all the logic related to deletedFlag, it's per secret version but all version ids have the same deletedFlag value True/False.
Is there any API to delete one/several versionId of a secret?  Otherwise, why not keep it to the secret_metadata table?

Reply:
will be remove deletedFlag .

Reply:
Do you plan to move it to the metadata table?